leader/src/rpc.rs

```diff
@@ -0,0 +1,199 @@
+use anyhow::Context;
+use ethers::types::{Address, Bloom, H256, U256};
+use evm_arithmetization::proof::BlockHashes;
+use futures::{stream::FuturesOrdered, TryStreamExt};
+use reqwest::IntoUrl;
+use serde::Deserialize;
+use tracing::info;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct EthGetBlockByNumberResult {
+    base_fee_per_gas: U256,
+    difficulty: U256,
+    gas_limit: U256,
+    gas_used: U256,
+    hash: H256,
+    logs_bloom: Bloom,
+    miner: Address,
+    mix_hash: H256,
+    number: U256,
+    parent_hash: H256,
+    state_root: H256,
+    timestamp: U256,
+
+    // Pre-eip-4895 blocks won't have this field, so if it's missing, using an empty vec is fine.
```

Comment:
I see you wanted to include it here as well :)

Reply:
Opps... This is honestly an oversight of `copy-and-paste`. :upside_down_face:

Nice catch, I'll remove this.

Reply:
Ah I see. Ok I think I did another commit after you did your review. All of these issues should be good at this point.